packages/@aws-cdk/aws-ec2/lib/instance.ts

```diff
@@ -361,6 +412,59 @@ export class Instance extends Resource implements IInstance {
     this.role.addToPolicy(statement);
   }
 
+  /**
+   * Use a CloudFormation Init configuration at instance startup
+   *
+   * This does the following:
+   *
+   * - Attaches the CloudFormation Init metadata to the Instance resource.
+   * - Add commands to the instance UserData to run `cfn-init` and `cfn-signal`.
+   * - Update the instance's CreationPolicy to wait for the `cfn-signal` commands.
+   */
+  private applyCloudFormationInit(init: CloudFormationInit, options: ApplyCloudFormationInitOptions = {}) {
+    const platform = this.osType === OperatingSystemType.WINDOWS ? InitPlatform.WINDOWS : InitPlatform.LINUX;
+    init._attach(this.instance, {
+      platform,
+      instanceRole: this.role,
+      userData: this.userData,
+      configSets: options.configSets,
+      embedFingerprint: options.embedFingerprint,
+      printLog: options.printLog,
+      ignoreFailures: options.ignoreFailures,
+    });
+    this.waitForResourceSignal(options.timeout ?? Duration.minutes(5));
+  }
+
+  /**
+   * Wait for a single additional resource signal
+   *
+   * Add 1 to the current ResourceSignal Count and add the given timeout to the current timeout.
+   *
+   * Use this to pause the CloudFormation deployment to wait for the instances
+   * in the AutoScalingGroup to report successful startup during
+   * creation and updates. The UserData script needs to invoke `cfn-signal`
+   * with a success or failure code after it is done setting up the instance.
+   */
+  private waitForResourceSignal(timeout: Duration) {
+    const oldResourceSignal = this.instance.cfnOptions.creationPolicy?.resourceSignal;
+    this.instance.cfnOptions.creationPolicy = {
+      ...this.instance.cfnOptions.creationPolicy,
+      resourceSignal: {
+        count: (oldResourceSignal?.count ?? 0) + 1,
+        timeout: (oldResourceSignal?.timeout ? Duration.parse(oldResourceSignal?.timeout).plus(timeout) : timeout).toIsoString(),
+      },
+    };
+  }
+
+  protected prepare() {
```

Comment:
Oops, can't have this anymore :(

https://github.com/aws/aws-cdk-rfcs/blob/master/text/0192-remove-constructs-compat.md#06-no-prepare-the-prepare-hook-is-no-longer-supported